src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java

```diff
@@ -468,9 +468,21 @@ public boolean checkoutFeatureBranch(Ref featureBranch) {
 			LOGGER.error("Commits cannot be retrieved since git object is null.");
 			return commitsForJiraIssue;
 		}
-		// @issue How to get the commits for branches that are not on the master branch?
-		// @decision Assume that the JIRA issue key equals the branch name, otherwise
-		// return commits on master branch
+		/**
+		 * @issue How to get the commits for branches that are not on the master branch?
+		 * @decision Assume that the branch name begins with the JIRA issue key!
+		 * @pro This simple rule will work just fine.
+		 * @con The rule is too simple. Issues with low key numbers could collect branches
+		 * of much higher issues also. ex. search for "CONDEC-1" would find branches
```

Comment:
We could overcome the problem that looking for CONDEC-1 could also return CONDEC-1x with the rule that there needs to be a colon, dot or space after the issue key. (The space is not allowed in branch names, but in commit messages.)

Reply:
You stated this below, sorry ;-)